compiler/riscv/codegen/FPTreeEvaluator.cpp

```diff
@@ -565,19 +567,95 @@ OMR::RV::TreeEvaluator::fRegLoadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return globalReg;
    }
 
+static TR::Register *
+commonFpMinMaxEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, bool reverse, bool isDouble, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT(node->getNumChildren() == 2, "The number of children for fmax/fmin/dmax/dmin must be 2.");
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *src2Reg = cg->evaluate(secondChild);
+
+   TR::Register *cmpReg = cg->allocateRegister();
+   TR::RealRegister *zero = cg->machine()->getRealRegister(TR::RealRegister::zero);
+   TR::Register *trgReg;
+
+   if (cg->canClobberNodesRegister(firstChild))
```

Comment:
Similarly to the integer min/max evaluator, you should be able to avoid duplicating the control flow logic and avoid the extra unconditional branch if the first child's node can't be clobbered by copying `src1Reg` to `trgReg` unconditionally before the control flow if needed. Unfortunately, there's no helper like `gprClobberEvaluate` for this case (yet), so you'll either need to add one or explicitly add the copy (if needed) here.

Reply:
Also fixed in bc26fcf.